docs/archive-redundancy.md

```diff
@@ -0,0 +1,63 @@
+3. Generate block data from another archive database: From a fully synced archive database, one can generate block data for each block using the `missing_subchain` tool. 
+
+    The tool takes an `--archive-uri`, a `--end-state-hash`, and an optional --start-state-hash and writes all the blocks in the chain starting from start-state-hash and ending at end-state-hash (including start and end).
+    If you provide both hashes, you get all blocks between start and end, and including the start and end. If just the end hash is provided, then it generates blocks starting with the unparented block closest to the end block, possibly the genesis block. The tool generates a file with name `<protocol-state-hash>.json` for each block. The block data in these files are called extensional blocks. Since these are generated from the database, they would have only the data stored in the archive database and would not contain any other information pertaining to a block (for example, blockchain snark)
+
+    Currently this tool is not packaged and can only be build from source using `make missing_subchain` 
+
+## Identifying missing blocks
+
+The tool `missing_block_auditor.exe` can be used to determine any missing blocks in an archive database. The tool outputs a list of state hashes of all the blocks that are missing. This can be used to monitor the archive database for any missing blocks.
+
+Currently this tool is not packaged and can only be build from source using `make missing_blocks_auditor`
+
+## Restoring blocks
+
+Missing blocks in an archive database can be added if there is block data (precomputed or extensional) available from the options listed in the backing-up-block-data section using the tool `mina-archive-blocks`. Can be built from source using `make archive_blocks`.
+
+1. For precomputed blocks: (Generated from option 1 and 2)
+
+    To import block data backed-up using option 1 and option 2 specify the `--precomputed` flag. It takes as arguments the paths of block data files that are to be imported
+
+2. For extensional blocks: (Generated from option 3)
+
+    To import extensional blocks generated from an archive database, specify the `--extensional` flag in the `archive_blocks` command
+
+## Staking ledgers
+
+Staking ledgers are used to determine slot winners for each epoch. Mina daemon stores staking ledger for the current epoch and the next epoch (after it is finalized). Whenever there is an epoch transition, the "next" staking ledger from the previous epoch is used to determine slot winners for the new epoch. The staking ledger from the previous epoch is replaced with the "next" staking ledger and a new "next" staking ledger is chosen. Since staking ledgers for older epochs are no longer accessible, users may want to still keep them around for reporting or other purposes.
+
+Currently these ledgers can be exported using the cli command:
+`coda.exe ledger export CURRENT-STAGED-LEDGER|STAKING-EPOCH-LEDGER|NEXT-EPOCH-LEDGER`
```

Comment:
the names of the ledgers need to be all-lowercase

I think it's a bug in the `help` system that prints them in uppercase

Reply:
the staged ledger is not a staking ledger, so maybe that should be described separately

also, the current syntax is just `staged-ledger`, not `current-staged-ledger`, and there's an optional `--state-hash` argument